Add documentation on updating SRE configurations (#2291)
alan-turing-institute/data-safe-haven: docs/source/management/index.md

All SREs associated with the SHM should be torn down before the SHM is torn down.
::::

> ### Updating SRE configurations

Reviewer: ```suggestion
### Updating SREs
```

I'm thinking a different working might make it clearer that we mean updating the SRE infrastructure and state/configuration, not just updating the configuration file.

> Changes to SRE configurations are made by first editing the relevant configuration file for the SRE, uploading the new configuration, and then redeploying the SRE.

Reviewer: ```suggestion
SREs are modified by updating the configuration then running the deploy command.
```

I don't like the original wording here. The configuration _is_ what is uploaded and stored in the context's storage account. There is no local configuration file that `dsh` will use.

- The existing configuration for the SRE can be shown using the following:

```{code} shell
$ dsh config show YOUR_SRE_NAME
```

> - If you do not have a local copy of the configuration for the SRE, it can be downloaded by adding the `file` argument:

Reviewer: ```suggestion
- If you do not have a local copy, you can write one with the `--file` option:
```

```{code} shell
$ dsh config show YOUR_SRE_NAME --file YOUR_SRE_NAME.yaml
```

- Edit the configuration file locally, and upload the new version:

```{code} shell
$ dsh config upload YOUR_SRE_NAME.yaml
```

- You will be shown the differences between the existing configuration and the new configuration and asked to confirm that they are correct.

> - Finally, redeploy your SRE for the infrastructure to reflect your new changes

Reviewer: ```suggestion
- You will be shown the differences between the existing configuration and the new configuration and asked to confirm that they are correct.
- Finally, deploy your SRE to apply any changes:
```

```{code} shell
$ dsh sre deploy YOUR_SRE_NAME
```

> ::::{admonition} Changing allowed administrator IP addresses

Reviewer: ```suggestion
::::{admonition} Changing administrator IP addresses
```

:class: warning

> If you are changing the IP addresses from which administrators are allowed to make changes to the infrastructure, you **must** redeploy the SRE from the **original** IP address.
> You will not be able to complete redeployment from the new IP address, because you will not be able to modify the IP addresses that are allowed to access the storage accounts.
> 
> - Add the new IP address to the configuration, but do not delete the original IP address.
> - Upload the new configuration and redeploy the SRE from the original IP address, as above
> 
> If you then wish to remove the original IP address from the list of allowed IP addresses, then
> 
> - Remove the old IP address from the configuration
> - Upload the new configuration and redeploy the SRE from the new IP address
> 

Reviewer: ```suggestion
The administrator IP addresses declared in the SRE configuration are used to create access rules for SRE infrastructure.
Therefore, after an SRE has been deployed, some changes can only be made from IP addresses on that list.

As a consequence, if you want to update the list of administrator IP addresses, for example to add a new administrator, you must do so from an IP address that is already allowed.
```

I think what you have written describes one particular problem rather than explaining the root cause.
